### PR TITLE
(#1646) Implementation of palindrome in C & C++ GSSoC'20

### LINKED ISSUE #1646: code to check if number is palindrome or not in C and C++
I want to work on this issue as a part of GSSoC'20.
Please assign me to this issue.

### PR DESCRIPTION
**Pull request by DawoodBhai420**
Fixes #1646 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- added C and C++ codes to check if a number or a string is palindrome or not.

#### Languages Used:
- C++
- C

#### Files Added:
- palindrome_no.c
- palindrome_no.cpp
- palindrome_string.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by DawoodBhai420** [comment]
@asha15  @raghav-dalmia @SakshiShreya 
can you please review the PR??

File: Palindrome/palindrome_no.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include<stdio.h>
+int main()
+{
+	int n,rem,temp,rev=0;
+	printf("\nEnter a number: ");
+	scanf("%d",&n);
+	temp=n;
+	while(temp!=0)
+	{
+		rem=temp%10;
```
Comment (MastersAbh) [inline Palindrome/palindrome_no.c]: Please follow 4 space indentation across the file. Here, it seems to have a "Tab"

File: Palindrome/palindrome_string.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#include<iostream>
+#include<string.h> // to use swap()
+using namespace std;
+void string_palindrome(string);
```
Comment (MastersAbh) [inline Palindrome/palindrome_string.cpp]: Do we need this declaration of this function here? If no, please remove it.

Comment (DawoodBhai420) [reply Palindrome/palindrome_string.cpp]: okay I will do that now

File: Palindrome/palindrome_no.c (hunk 1 above)
Comment (DawoodBhai420) [reply Palindrome/palindrome_no.c]: @MastersAbh  can you please tell me how to do so??

**Comment by DawoodBhai420** [comment]
@MastersAbh  made required changes please review.

File: Palindrome/palindrome_no.c (hunk 1 above)
Comment (MastersAbh) [reply Palindrome/palindrome_no.c]: Instead of tab as an indentation, please add 4 spaces for each block

**Comment by DawoodBhai420** [comment]
made requested changes.
please review.

**Comment by DawoodBhai420** [comment]
Thank you !!

**Pull request merged**